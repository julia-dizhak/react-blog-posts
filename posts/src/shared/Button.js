import React from 'react';

import PropTypes from 'prop-types';

import withLoading from './../hoc/withLoading';

const propTypes = {
    onClick: PropTypes.func.isRequired,
    className: PropTypes.string,
    children: PropTypes.node.isRequired
}, defaultProps = {
    className: 'button'
};

export function Button(props) {
    const {className, onClick, children} = props;
    
    return (
        <button 
           className={className}
           onClick={onClick}>
            {children}
        </button>
    );
}  

export const ButtonWithLoading = withLoading(Button);

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
