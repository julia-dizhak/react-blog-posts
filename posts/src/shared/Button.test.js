import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'; 

import {Button} from './Button';

describe('Button component', () => {
    it('renders without crashing', () => {
        const props = { onClick: () => {} };
        const div = document.createElement('div');

        ReactDOM.render(<Button {...props}>Give Me More</Button>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    test('has a valid snapshot', () => {
        const props = { onClick: () => {} };
        const component = renderer.create(<Button {...props}>Give Me More</Button>);
        
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
