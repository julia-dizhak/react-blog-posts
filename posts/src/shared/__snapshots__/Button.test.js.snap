// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component has a valid snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Give Me More
</button>
`;
